### PR TITLE
Partial ES 6 & 7.

### PR DESCRIPTION
**Pull request by diasbruno**
This implements features for es 6 & 7.

- [x] const and let
- [x] import
- [x] export
- [x] arrows
- [ ] default arguments
- [x] object matching (destructuring, single name bind)
- [ ] rest + spread
- [ ] classes + extends
- [ ] template strings

Issue #59.

### CODE REVIEW

**Comment by diasbruno** [comment]
I'll leave this PR open so everyone can help me. :)

File: test/Test/Language/Javascript/StatementParser.hs [hunk 1]
```diff
@@ -23,8 +23,21 @@ testStatementParser = describe "Parse statements:" $ do
         testStmt "{{}}"         `shouldBe` "Right (JSAstStatement (JSStatementBlock [JSStatementBlock []]))"
         testStmt "{{{}}}"       `shouldBe` "Right (JSAstStatement (JSStatementBlock [JSStatementBlock [JSStatementBlock []]]))"
 
-    it "if" $
+    it "if" $ do
         testStmt "if (1) {}"    `shouldBe` "Right (JSAstStatement (JSIf (JSDecimal '1') (JSStatementBlock [])))"
+        -- fix: fix ambiguity with block and object literal on if block position
+        -- testStmt "if (1) {x}"    `shouldBe` "Right (JSAstStatement (JSIf (JSDecimal '1') (JSStatementBlock [JSIdentifier 'x'])))"
```
Comment (diasbruno) [inline test/Test/Language/Javascript/StatementParser.hs:29]: @erikd I need some help with this. I still can't figure it out a way to disambiguiate block and object on a `if statement`. The minified version would print something like:

`if(a){{x}}...`

Comment (diasbruno) [reply test/Test/Language/Javascript/StatementParser.hs:29]: The inner `{}` is the object.

Comment (diasbruno) [reply test/Test/Language/Javascript/StatementParser.hs:29]: Probably this happens because it will try ObjectLiteral first (?).

Comment (erikd) [reply test/Test/Language/Javascript/StatementParser.hs:29]: I'm fine with minor sensible changes to the tests as needed.

**Comment by pedrofurla** [comment]
I am in need of arrow functions support, is there anything I can do to move help move this forward?

**Comment by diasbruno** [comment]
@pedrofurla Sure. If you can make progress with this PR...please. I didn't have much time to finish it. I wrote [language-js](https://github.com/diasbruno/language-js) just for what I needed, but I hope this one can be finished as well.

**Comment by pedrofurla** [comment]
To be honest I am happy with its current state, but I will give it a go tomorrow and see what I can do. I make not promises. :)

**Comment by dcastro** [comment]
@diasbruno, thank you so much for looking into this!

I tried pulling this branch to build my project, and I'm getting this error:

> src/Language/JavaScript/Process/Minify.hs:(133,5)-(160,100): Non-exhaustive patterns in function fix

Looks like the [`MinifyJS` instance for `JSExpression`](https://github.com/diasbruno/language-javascript/blob/59dbe22d5a7b235a686134a62567b6de9e55e8af/src/Language/JavaScript/Process/Minify.hs#L131-L160) is missing a case for `JSArrowExpression` and `JSSpreadExpression`?

**Comment by diasbruno** [comment]
@dcastro I've implemented just the 'parser' part and few things just to compile/run tests.

**Comment by michaelficarra** [comment]
@diasbruno Do you think it would be better to split off the unimplemented features into a separate effort and get this PR into a merge-able state? Partial support of new features is definitely better than no support.

**Comment by erikd** [comment]
I'd be find with partial support.

**Comment by diasbruno** [comment]
That would be great. I was begin to worry about not getting time to finish this. 
@erikd Please, let me know how can we get this 'first part' done.

Thank you all for the help.

**Comment by erikd** [comment]
I would like to do a really through review of this before merging. Hopefully get around to this on the weekend.

**Comment by erikd** [comment]
Going to have to revert this from the `new-ast` branch for now. Its not complete. There are incomplete pattern matches in the pretty printer and minifier.

Will look at this myself today.

**Pull request merged**